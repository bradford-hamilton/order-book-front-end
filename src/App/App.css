@import url('https://fonts.googleapis.com/css?family=Raleway');

body {
  background-color: #F2F2F0;
  font-family: 'Raleway', sans-serif;
}

.headerRow {
  text-align: center;
  padding-bottom: 30px;
}

.getStartedRow {
  text-align: center;
}

.App h1 {
  font-size: 4rem;
  padding-top: 10px;
  padding-bottom: 15px;
}

.App h4 {
  font-size: 2rem;
  color: grey;
}

.market {
  text-align: center;
  margin-top: 15px;
}

.orderBooks {
  padding-top: 30px;
  padding-bottom: 30px;
}

.bookBackground {
  background-color: #daedff;
  padding: 4px;
  border-radius: 4px;
  box-shadow: 1px 1px 2px 2px #B0BCCB;
}

.orderColumns {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.orderWrapper {
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin-bottom: 4px;
  background-color: #FFFFFF;
  height: 30px;
  width: 100%;
  border-radius: 3px;
}

.bid {
  margin: 0;
  color: green;
  width: 100px;
}

.ask {
  margin: 0;
  color: red;
  width: 100px;
}

.exchange {
  margin: 0;
  width: 100px;
  text-align: right;
}

.quantity {
  margin: 0;
  color: grey;
  width: 100px;
}
