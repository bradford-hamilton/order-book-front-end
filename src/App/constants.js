export const MARKETS = [
  'BTC-ETH',
  'BTC-LTC',
  'BTC-XMR',
  'BTC-DASH',
  'BTC-DOGE',
];
