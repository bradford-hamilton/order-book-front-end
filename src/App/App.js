import React, { Component } from 'react';
import socketIOClient from 'socket.io-client'
import { Grid, Row, Col, Button, MenuItem, DropdownButton } from 'react-bootstrap';
import NavBar from './NavBar/NavBar';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      socket: socketIOClient('http://localhost:4000'),
      response: null,
      loading: true,
      selected: 'BTC-ETH',
    };
  }

  componentDidMount() {
    this.state.socket.emit('marketPair', this.state.selected);
    this.state.socket.on('newData', data => {
      this.setState({
        response: data,
        loading: false,
      });
    });
  }

  send = (marketPair) => {
    this.state.socket.emit('marketPair', marketPair);
    this.setState({
      loading: true,
      selected: marketPair,
    });
  }

  render() {
    console.log(this.state.response);
    return (
      <div className="App">
        <NavBar />
        <Grid>
          <Row className="headerRow">
            <Col xs={12}>
              <h1>Welcome to the MasterBook</h1>
              <h4>May the odds be ever in your favour.</h4>
            </Col>
          </Row>
          <Row className="getStartedRow">
            <Col xs={12} md={6} mdOffset={3}>
              <DropdownButton
                bsStyle="primary"
                title="Select Market"
                id="dropdown-basic"
                onSelect={this.send}
              >
                <MenuItem
                  eventKey="BTC-ETH"
                  active={this.state.selected === 'BTC-ETH'}
                >
                  BTC-ETH
                </MenuItem>
                <MenuItem
                  eventKey="BTC-DOGE"
                  active={this.state.selected === 'BTC-DOGE'}
                >
                  BTC-DOGE
                </MenuItem>
              </DropdownButton>
            </Col>
          </Row>
          <Row>
            <Col xs={12} className="market">
              <h3>Selected Market: {this.state.selected}</h3>
            </Col>
          </Row>
          <Row className="orderBooks">
            <Col xs={12} md={6}>
              <div className="orderColumns">
                <p>Bid</p>
                <p>Amount</p>
                <p>Exchange</p>
              </div>
              {this.state.loading ? (
                <div>loading</div>
              ) : (
                <div className="bookBackground">
                  {this.state.response && (
                    this.state.response.allBids.map((bid, i) => (
                      <div className="orderWrapper" key={i}>
                        <p className="bid">{bid.rate.toFixed(8)}</p>
                        <p className="quantity">{bid.quantity.toFixed(8)}</p>
                        <p className="exchange">{bid.exchange}</p>
                      </div>
                    ))
                  )}
                </div>
              )}
            </Col>
            <Col xs={12} md={6}>
              <div className="orderColumns">
                <p>Ask</p>
                <p>Amount</p>
                <p>Exchange</p>
              </div>
              {this.state.loading ? (
                <div>loading</div>
              ) : (
                <div className="bookBackground">
                  {this.state.response && (
                    this.state.response.allAsks.map((ask, i) => (
                      <div className="orderWrapper" key={i}>
                        <p className="ask">{ask.rate.toFixed(8)}</p>
                        <p className="quantity">{ask.quantity.toFixed(8)}</p>
                        <p className="exchange">{ask.exchange}</p>
                      </div>
                    ))
                  )}
                </div>
              )}
            </Col>
          </Row>
        </Grid>
      </div>
    );
  }
}

export default App;
