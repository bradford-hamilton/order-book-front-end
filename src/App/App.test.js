import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import App from './App';

configure({ adapter: new Adapter() });

function setup() {
  const comp = shallow(<App />);

  return { comp };
}

describe('<App />', () => {
  it('renders without crashing', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
