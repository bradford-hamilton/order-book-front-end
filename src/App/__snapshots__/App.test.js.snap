// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> snapshot 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-inverse"
  >
    <div
      className="container"
    >
      <div
        className="navbar-header"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          MasterBook
        </a>
      </div>
    </div>
  </nav>
  <div
    className="container-fluid"
  >
    <div
      className="headerRow row"
    >
      <div
        className="col-xs-12"
      >
        <h1>
          Welcome to the MasterBook
        </h1>
        <h4>
          May the numbers be ever in your favour.
        </h4>
      </div>
    </div>
    <div
      className="getStartedRow row"
    >
      <div
        className="col-md-6 col-md-offset-3 col-xs-12"
      >
        <div
          className="dropdown btn-group btn-group-primary"
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-primary"
            disabled={false}
            id="dropdown-basic"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            type="button"
          >
            Select Market
             
            <span
              className="caret"
            />
          </button>
          <ul
            aria-labelledby="dropdown-basic"
            className="dropdown-menu"
            role="menu"
          >
            <li
              className="active"
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                BTC-ETH
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                BTC-LTC
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                BTC-XMR
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                BTC-DASH
              </a>
            </li>
            <li
              className=""
              role="presentation"
              style={undefined}
            >
              <a
                href="#"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="menuitem"
                tabIndex="-1"
              >
                BTC-DOGE
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="market col-xs-12"
      >
        <h3>
          Selected Market: 
          BTC-ETH
        </h3>
      </div>
    </div>
    <div
      className="container"
    >
      <div
        className="orderBooks row"
      >
        <div
          className="col-md-6 col-xs-12"
        >
          <h4
            className="orderBookType"
          >
            <span
              className="bold"
            >
              Live Order Book
            </span>
             - 
            Bid
          </h4>
          <div
            className="orderColumns"
          >
            <p>
              Bid
            </p>
            <p>
              Amount
            </p>
            <p>
              Exchange
            </p>
          </div>
          <div
            className="ringLoader"
          >
            <div
              className="css-jxiqlq"
            >
              <div
                className="css-asv35b"
              />
              <div
                className="css-1ukn1c"
              />
            </div>
          </div>
        </div>
        <div
          className="col-md-6 col-xs-12"
        >
          <h4
            className="orderBookType"
          >
            <span
              className="bold"
            >
              Live Order Book
            </span>
             - 
            Ask
          </h4>
          <div
            className="orderColumns"
          >
            <p>
              Ask
            </p>
            <p>
              Amount
            </p>
            <p>
              Exchange
            </p>
          </div>
          <div
            className="ringLoader"
          >
            <div
              className="css-jxiqlq"
            >
              <div
                className="css-asv35b"
              />
              <div
                className="css-1ukn1c"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
